### ISSUE TITLE
Tree control does not always create items with the passed root 

### ISSUE THREAD

**Issue by RodeoMcCabe** [godotengine/godot#12118]
Linux, Debian 8
Godot 3, build bf5a927 (Oct 14, from calinou's nightly builds)

Tree.create_item(parent) doesn't always create the item with the passed parent. It does not behave as described in the docs (which it should, the behavior described in the docs makes sense).

**Steps to reproduce:**
```
var root = tree.create_item() # becomes root, as expected

var section1 = tree.create_item(root) # become child of root, as expected

var item1 = tree.create_item(section1) # becomes actual root and the var root becomes child of item1, not expected
var item2 = tree.create_item(section1) # becomes actual root and item1 becomes child of item2, not expected

var section2 = tree.create_item(root) # becomes child of the var root, as expected, but var root is no longer the actual root of the tree

var item3 = tree.create_item(section2) # becomes actual root and item2 becomes child of item3, not expected

var no_parent = tree.create_item() # becomes actual root and item3 becomes child of no_parent, not expected
```

**Link to minimal example project:**
[github](https://github.com/RodeoMcCabe/godot-tree-bug)
[tree-test.gz](https://github.com/godotengine/godot/files/1385165/tree-test.gz)
[tree-test.zip](https://github.com/godotengine/godot/files/1385166/tree-test.zip)

**Comment by RodeoMcCabe** [comment]
I'm working on fixing this now.

**Comment by RayKoopa** [comment]
Waiting for this to be fixed. I don't need the new optional insertion parameter anymore, but if it's of interest, I'd add it after this is merged.

**Comment by RodeoMcCabe** [comment]
I somewhat agree with reduz. On the other hand, I think if calling the method without an argument sometimes throws and error and sometimes doesn't, that could be confusing too. I'm not sure what the best approach would be.

In any case, I'm not working on this anymore as I've found a workaround for now. If you'd like to work on it feel free, you can look at the [PR I made](https://github.com/godotengine/godot/pull/13725#issuecomment-352185382). I explain there how I worked around it.

**Comment by akien-mga** [comment]
Fixed by #14801.
